### PR TITLE
Feature/improve docs

### PR DESCRIPTION
**Pull request by pravinbhat**
Improved docs for the properties file
Set default max-retry to 0 & updated applicable logging

### CODE REVIEW

File: src/main/java/datastax/astra/migrate/OriginCountJobSession.java [hunk 1]
```diff
@@ -77,9 +77,9 @@ public static OriginCountJobSession getInstance(CqlSession sourceSession, SparkC
 
     public void getData(BigInteger min, BigInteger max) {
         logger.info("ThreadID: {} Processing min: {} max: {}", Thread.currentThread().getId(), min, max);
-        int maxAttempts = maxRetries;
-        for (int retryCount = 1; retryCount <= maxAttempts; retryCount++) {
-
+        boolean done = false;
+        int maxAttempts = maxRetries + 1;
```
Comment (msmygit) [inline src/main/java/datastax/astra/migrate/OriginCountJobSession.java:81]: #nitpick could we simplify it by removing an extra `maxAttempts` variable which is just used once in the for loop?

```
for (int attempts = 1; attempts <= (maxRetries + 1) && !done; attempts++) {
```

Same change for `DiffJobSession.java` and `CopyJobSession.java` as well.

File: src/main/java/datastax/astra/migrate/AbstractJobSession.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ protected AbstractJobSession(CqlSession sourceSession, CqlSession astraSession,
 
         readLimiter = RateLimiter.create(new Integer(Util.getSparkPropOr(sc, "spark.readRateLimit", "20000")));
         writeLimiter = RateLimiter.create(new Integer(Util.getSparkPropOr(sc, "spark.writeRateLimit", "40000")));
-        maxRetries = Integer.parseInt(sc.get("spark.maxRetries", "10"));
+        maxRetries = Integer.parseInt(sc.get("spark.maxRetries", "0"));
```
Comment (msmygit) [inline src/main/java/datastax/astra/migrate/AbstractJobSession.java:45]: curious, why are we removing the default retry mechanism? Is it taking too long?

**Review by msmygit — APPROVED** [review]
LGTM. I've some trivial nitpicks.

File: src/main/java/datastax/astra/migrate/OriginCountJobSession.java (hunk 1 above)
Comment (pravinbhat) [reply src/main/java/datastax/astra/migrate/OriginCountJobSession.java:81]: @msmygit It was added for code clarity but can be removed if you think its adding to confusion. Retries are now disabled by default & the logs report every attempt when it does not work

File: src/main/java/datastax/astra/migrate/AbstractJobSession.java (hunk 2 above)
Comment (pravinbhat) [reply src/main/java/datastax/astra/migrate/AbstractJobSession.java:45]: @msmygit Based on experience, retries rarely help & when it does, we usually end up reporting incorrect stats. The stats issue is difficult to address as when a particular partition-range fails, we do not know how many records it actually migrated before failing during each attempt. We will need complex logic to track this for every attempt.

Earlier we preferred retires as we did not have an option to rerun only the failed partitions & had to rerun the full migration again, but we added feature to migrate only failed partitions https://github.com/datastax/cassandra-data-migrator#migrating-specific-partition-ranges & that way we always report accurate stats.

**Pull request merged**